grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -205,29 +211,50 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
                 private HttpData content = HttpData.empty();
                 @Nullable
                 private HttpHeaders trailers;
+                @Nullable
+                private Subscription subscription;
+                @Nullable
+                private Throwable cause;
                 private int processedMessages;
 
                 @Override
                 public void onSubscribe(Subscription s) {
+                    if (subscription != null) {
+                        s.cancel();
+                        return;
+                    }
```

Comment:
Is it possible for this method to get invoked more than once? If not, how about just adding an assertion?
```suggestion
                    assert subscription == null;
```

Reply:
I'm following the reactive streams spec https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.3/README.md#specification:

> A Subscriber MUST call Subscription.cancel() on the given Subscription after an onSubscribe signal if it already has an active Subscription.

> Calling onSubscribe, onNext, onError or onComplete MUST return normally except when any provided parameter is null in which case it MUST throw a java.lang.NullPointerException to the caller, for all other situations the only legal way for a Subscriber to signal failure is by cancelling its Subscription.

Reply:
TIL :bow: but do we actually expect that to happen? In our case that'd be a bug if it ever happens IIUC.

Reply:
> The intent of this rule is to prevent that two, or more, separate Publishers from trying to interact with the same Subscriber. Enforcing this rule means that resource leaks are prevented since extra Subscriptions will be cancelled. Failure to conform to this rule may lead to violations of Publisher rule 1, amongst others. Such violations can lead to hard-to-diagnose bugs.

There's no harm in sticking with the spec. If you'd like, I can add warning logging to make sure this issue is surfaced if it happens.